### PR TITLE
lib25519: 20240321 -> 20241004, add update script, add version test script

### PR DESCRIPTION
**Pull request by kiike**
- include #339581
- 20240321 -> 20241004
- add version test script

[Changelog](https://lib25519.cr.yp.to/download.html)

Closes: https://github.com/NixOS/nixpkgs/pull/339581

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Adda0** [comment]
Does this mean that https://github.com/NixOS/nixpkgs/pull/339581 can be closed (to not needlessly duplicate the same PR content)? I presume that is the case.

File: pkgs/by-name/li/lib25519/package.nix [hunk 1]
```diff
@@ -31,7 +37,10 @@ stdenv.mkDerivation (finalAttrs: {
     runHook postConfigure
   '';
 
-  nativeBuildInputs = [ python3 ];
+  nativeBuildInputs = [
+    python3
+    valgrind
```
Comment (Adda0) [inline pkgs/by-name/li/lib25519/package.nix:42]: Why is `valgrind` necessary?

**Comment by kiike** [comment]
> Does this mean that #339581 can be closed (to not needlessly duplicate the same PR content)? I presume that is the case.

Yes! The idea is integrate both changes.

File: pkgs/by-name/li/lib25519/package.nix (hunk 1 above)
Comment (kiike) [reply pkgs/by-name/li/lib25519/package.nix:42]: It's a build dependency of the testing component of the project.

Comment (Adda0) [reply pkgs/by-name/li/lib25519/package.nix:42]: Thanks for the explanation.

**Review by Adda0 — APPROVED** [review]
LGTM. It would be great if someone else could approve the PR, too, however.

**Review by imadnyc — APPROVED** [review]

**Comment by misuzu** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 350211`

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>lib25519</li>
  </ul>
</details>

**Pull request merged**